hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -945,23 +945,33 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 	hcp.Annotations = map[string]string{
 		hostedClusterAnnotation: client.ObjectKeyFromObject(hcluster).String(),
 	}
-	for annotationKey := range hcluster.Annotations {
-		if annotationKey == hyperv1.DisablePKIReconciliationAnnotation {
-			hcp.Annotations[hyperv1.DisablePKIReconciliationAnnotation] = hcluster.Annotations[hyperv1.DisablePKIReconciliationAnnotation]
-		} else if annotationKey == hyperv1.OauthLoginURLOverrideAnnotation {
-			hcp.Annotations[hyperv1.OauthLoginURLOverrideAnnotation] = hcluster.Annotations[hyperv1.OauthLoginURLOverrideAnnotation]
-		} else if strings.HasPrefix(annotationKey, hyperv1.IdentityProviderOverridesAnnotationPrefix) {
-			hcp.Annotations[annotationKey] = hcluster.Annotations[annotationKey]
-		} else if annotationKey == hyperv1.KonnectivityAgentImageAnnotation || annotationKey == hyperv1.KonnectivityServerImageAnnotation {
-			hcp.Annotations[annotationKey] = hcluster.Annotations[annotationKey]
-		} else if annotationKey == hyperv1.RestartDateAnnotation {
-			hcp.Annotations[annotationKey] = hcluster.Annotations[annotationKey]
-		} else if annotationKey == hyperv1.IBMCloudKMSProviderImage || annotationKey == hyperv1.AWSKMSProviderImage {
-			hcp.Annotations[annotationKey] = hcluster.Annotations[annotationKey]
-		} else if annotationKey == hyperv1.PortierisImageAnnotation {
-			hcp.Annotations[hyperv1.PortierisImageAnnotation] = hcluster.Annotations[hyperv1.PortierisImageAnnotation]
+
+	// These annotations are copied from the HostedCluster
+	mirroredAnnotations := []string{
+		hyperv1.DisablePKIReconciliationAnnotation,
+		hyperv1.OauthLoginURLOverrideAnnotation,
+		hyperv1.KonnectivityAgentImageAnnotation,
+		hyperv1.KonnectivityServerImageAnnotation,
+		hyperv1.RestartDateAnnotation,
+		hyperv1.IBMCloudKMSProviderImage,
+		hyperv1.AWSKMSProviderImage,
+		hyperv1.PortierisImageAnnotation,
+		hyperutil.DebugDeploymentsAnnotation,
+	}
+	for _, key := range mirroredAnnotations {
```

Comment:
Nit: You could range over the hcluster annotations and copy them to the hcp if they are mirrored or have the prefix, saving yourself one for loop (would require mirroredAnnotations to be a Set)

Reply:
True, although I explicitly separated out the prefix bit because it seemed like a special case and worth keeping clearly apart; optimization at this level doesn't seem important at all vs. clarity IMO